logger.c

```diff
@@ -371,6 +371,59 @@ static int _logger_parse_prx_req(logentry *e, char *scratch) {
             );
     return total;
 }
+
+static void _logger_log_proxy_errbe(logentry *e, const entry_details *d, const void *entry, va_list ap) {
+    char *errmsg = va_arg(ap, char *);
+    char *be_name = va_arg(ap, char *);
+    char *be_port = va_arg(ap, char *);
+    char *be_rbuf = va_arg(ap, char *);
+    int be_rbuflen = va_arg(ap, int);
+
+    struct logentry_proxy_errbe *le = (void *)e->data;
+    le->errlen = strlen(errmsg);
+    if (be_name && be_port) {
+        le->be_namelen = strlen(be_name);
+        le->be_portlen = strlen(be_port);
+    }
+
+    le->be_rbuflen = be_rbuflen;
+    if (be_rbuflen > 100) {
+        le->be_rbuflen = 100;
+    }
+
+    char *data = le->data;
+    memcpy(data, errmsg, le->errlen);
+    data += le->errlen;
+    memcpy(data, be_name, le->be_namelen);
+    data += le->be_namelen;
+    memcpy(data, be_port, le->be_portlen);
+    data += le->be_portlen;
+    memcpy(data, be_rbuf, le->be_rbuflen);
+}
+
+#define MAX_RBUF_READ 300
+static int _logger_parse_prx_errbe(logentry *e, char *scratch) {
+    int total;
+    char rbuf[MAX_RBUF_READ];
+    struct logentry_proxy_errbe *le = (void *)e->data;
+    char *data = le->data;
+    char *errmsg = data;
+    data += le->errlen;
+    char *be_name = data;
+    data += le->be_namelen;
+    char *be_port = data;
+    data += le->be_portlen;
+    char *be_rbuf = data;
+
+    uriencode(be_rbuf, rbuf, le->be_rbuflen, MAX_RBUF_READ);
+    total = snprintf(scratch, LOGGER_PARSE_SCRATCH,
```

Comment:
Would it be helpful to log some from the end as well?

Reply:
Not that I can think of, no. It's almost always going to be an error message or some obvious garbage, and in all cases I've seen it's been short data anyway.

Reply:
Ack